app/main/updater.js

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import os from 'os';
+import { setInterval } from 'timers';
+import { app, autoUpdater, dialog } from 'electron';
+import { version as appVersion, name as appName } from '../../package.json';
+
+// Check once an hour
+const UPDATE_CHECK_TIME = 1000 * 60 * 60;
+
+// Exe name is set to the same as the package name
+app.setAppUserModelId(`com.squirrel.${appName}.${appName}`);
+
+try {
+  autoUpdater.setFeedURL(`https://tofino-update-server.herokuapp.com/update/${os.platform()}_${os.arch()}/${appVersion}`);
+} catch (e) {
+  // This happens when the application isn't code signed on OSX.
+}
+
+autoUpdater.on('update-downloaded', () => {
+  dialog.showMessageBox({
+    type: 'question',
```

Comment:
See the way setting the default browser works.

1. main process sends ipc message to renderer process "should-quit-and-update"
2. renderer process receives ipc message and dispatches "CONFIRM_QUIT_AND_UPDATE" effect
3. said effect is handled in the appropriate ui saga
4. saga sends ipc message to main process "show-message-box" with the appropriate details (this will later be replaced with better ui; it also has the benefit of standardizing a way of showing arbitrary message boxes from the ui)
5. upon confirmation, send ipc message to renderer process "${message-box-id}-confirmed" or something similar; if dismissed or cancelled, "${message-box-id}-cancelled"
6. the above mentioned messages are handled in the appropriate ui saga

This might involve a lot of boilerplate, but is very testable and easily changed in the future when we decide to use different/better ui.